compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -471,7 +475,8 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
                                 // trigger cycle errors if we try to compute the value of the other static
                                 // and that static refers back to us (potentially through a promoted).
                                 // This could miss some UB, but that's fine.
-                                skip_recursive_check = true;
+                                // We still walk nested allocations, as they are fundamentally part of this validation run.
+                                skip_recursive_check = !nested;
```

Comment:
What if this is a nested static of *another* static? Then we'd recurse but we should not.

I think it's fine though, it just means we do a bit of redundant work. But seems worth a comment.

```suggestion
                                // This means we will also recurse into nested statics of *other*
                                // statics. That's somewhat inconsistent but harmless.
                                skip_recursive_check = !nested;
```

Reply:
> The only reason we don't recurse into other statics is cycle errors.

And performance, potentially.

Reply:
Note that this preserves the previous behaviour, we didn't know an allocation was a nested one of a static and just walked it.

Reply:
Yes, I think it's fine to preserve.